src/disp_s1/browse_image.py

```diff
@@ -18,18 +101,64 @@ def make_browse_image(
     ----------
     output_filename : Filename
         Name of output PNG
-    arr : ArrayLike
-        input 2D image array
+    input_filename : Filename
+        Name of input NetCDF file.
+    dataset_name: str
+        Name of datast to be made into a browse image.
     max_dim_allowed : int, default = 2048
         Size (in pixels) of the maximum allowed dimension of output image.
         Image gets rescaled with same aspect ratio.
     """
-    orig_shape = arr.shape
-    scaling_ratio = max([s / max_dim_allowed for s in orig_shape])
-    # scale original shape by scaling ratio
-    scaled_shape = [int(np.ceil(s / scaling_ratio)) for s in orig_shape]
+    # check if dataset can be plotted
+    valid_dataset_names = [
+        "unwrapped_phase",
+        "connected_component_labels",
+        "temporal_coherence",
+        "interferometric_correlation",
+        "persistent_scatterer_mask",
+    ]
+    if dataset_name not in valid_dataset_names:
+        raise ValueError(f"{args.dataset_name} is not a valid dataset name")
+
+    # get dataset as array from input NC file
+    with h5netcdf.File(input_filename, "r") as nc_handle:
+        arr = nc_handle[dataset_name][()]
+
+    # nomalize non-nan pixels to 0-1
+    arr = _normalize_apply_gamma(arr)
+
+    # compute browse shape and resize full size array to it
+    arr = _resize_to_max_pixel_dim(arr, max_dim_allowed)
+
+    _save_to_disk_as_greyscale(arr, output_filename)
+
+
+if __name__ == "__main__":
```

Comment:
also thinking for future- we might end up making the `disp-s1` command line tool have sub commands, in which case this would be nice to have as `disp-s1 browse ...`. But that'll require more setup